src/main/kotlin/pl/org/pablo/slack/money/slack/parser/ArgumentParser.kt

```diff
@@ -0,0 +1,31 @@
+package pl.org.pablo.slack.money.slack.parser
+
+import org.springframework.stereotype.Component
+
+interface Parser<out R> {
+    fun canParse(command: String): Boolean
+    fun canFinalize(): Boolean
+    fun parse(command: String): Boolean
+    fun finalize(): R
+}
+
+interface ArgumentParser {
+    fun <R> parse(parser: Parser<R>, command: String): R
+}
+
+@Component
+class ArgumentParserImpl : ArgumentParser {
+    override fun <R> parse(parser: Parser<R>, command: String): R {
+        val commandList = (command.split(" ").toList())
```

Comment:
The parenthesis is not required arround (command ...). Also maybe call it "commands" ;)

Reply:
Now `command` is whole text received from slack and `element` is single element of that command.